sentry-ruby/lib/sentry/transport.rb

```diff
@@ -82,6 +82,22 @@ def serialize_envelope(envelope)
           result = item.to_s
         end
 
+        if result.bytesize > Event::MAX_SERIALIZED_PAYLOAD_SIZE
+          if single_exceptions = item.payload.dig(:exception, :values)
+            single_exceptions.each do |single_exception|
+              traces = single_exception.dig(:stacktrace, :frames)
+              traces.replace(traces[-10..-1]) if traces && traces.size > 10
+            end
+          elsif single_exceptions = item.payload.dig("exception", "values")
+            single_exceptions.each do |single_exception|
+              traces = single_exception.dig("stacktrace", "frames")
+              traces.replace(traces[-10..-1]) if traces && traces.size > 10
+            end
+          end
+
+          result = item.to_s
+        end
+
```

Comment:
As mentioned in https://github.com/getsentry/sentry-ruby/pull/1806#discussion_r863590942, we should update the log message as well (see line 106).

Reply:
What should it be updated to since this PR remove some frames only (or keeping some frames)